LocalWiki.php

```diff
@@ -81,10 +82,19 @@ function onHtmlPageLinkRendererEnd(
 				return true;
 			}
 
-			$attribs['rel'] = 'nofollow';
+			$attribs[ 'rel' ] = 'nofollow';
 
 			return true;
 		}
+		
+		function onPersonalUrls( array &$personal_urls, Title $title, SkinTemplate $skin ) {
+			$personal_urls[ 'preferences' ] = [
+				'text' => 'Preferences',
+				'href' => 'https://genshinimpact.miraheze.org/wiki/Genshin_Impact_Wiki:Preferences'
+			];
+		}
+		
+		$wgAllowSiteCSSOnRestrictedPages = true;
```

Comment:
This is a huge no-no. This is not going to be enabled.

Reply:
Ok, I removed it, but why? It is only CSS, not JS.

Reply:
Even if it's only CSS you can still do nasty things. I'd rather be safe than sorry.

Reply:
[https://phabricator.wikimedia.org/T72672 I see]. Is there any way to implement some of my custom CSS on the login and create account pages without enabling CSS on restricted pages altogether?

Reply:
No.

Reply:
Actually, what if I added a hook in LocalWiki to add CSS to the login page's head tag? That way, the CSS would have to be reviewed before it can be added to the login page.